### ISSUE TITLE
YOLOv8n-cls - Classify ignores batches = slow train time and idle GPU freqs

### ISSUE THREAD

**Issue by ArsBinarii** [ultralytics/ultralytics#10083]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

YOLOv8n-cls - Classify ignores batches = slow train time and idle GPU freqs
takes only 1GB of GPU mem no matter the size of the batch

### Environment

Ultralytics YOLOv8.1.35 🚀 Python-3.9.19 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4080, 16375MiB)
Setup complete ✅ (20 CPUs, 64.0 GB RAM, 100.2/120.0 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.9.19
Install             pip
RAM                 64.00 GB
CPU                 AMD Ryzen 9 5900X 12-Core Processor
CUDA                12.1

matplotlib          ✅ 3.8.3>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.12.0>=1.4.1
torch               ✅ 2.2.1+cu121>=1.8.0
torchvision         ✅ 0.17.1+cu121>=0.9.0
tqdm                ✅ 4.66.2>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.1>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0

### Minimal Reproducible Example

YOLOv8n-cls summary: 99 layers, 2719288 parameters, 2719288 gradients, 4.4 GFLOPs
New https://pypi.org/project/ultralytics/8.1.47 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.1.35 🚀 Python-3.9.19 torch-2.2.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4080, 16375MiB)
engine\trainer: task=classify, mode=train, model=yolov8n-cls.yaml, data=.\data\, epochs=100, time=None, patience=100, batch=4096, imgsz=224, save=True, save_period=-1, cache=True, device=None, workers=10, project=None, name=train6, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\classify\train6
train: F:\yolo-font\data\train... found 5423200 images in 1205 classes ✅
val: F:\yolo-font\data\val... found 1352423 images in 1205 classes ✅

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@ArsBinarii thanks for updating us! It's great to hear that the issue was resolved in Ultralytics YOLOv8.1.47 🚀. If you encounter any more questions or run into other issues, feel free to reach out. Happy coding!
